### PR TITLE
go go gadget hat

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

adds this, a hat that you can say a phrase for it to put something in your hands

https://github.com/tgstation/tgstation/assets/70376633/5b4c86fd-8568-4cbf-9b4d-215bc55e94da

video explains use

## Why It's Good For The Game

funny

## Changelog
:cl:
add: added the inspectors hat to the detectives cabinet, a special hat that allows the wearer to say a phrase to dispense a stored item
/:cl:

### CODE REVIEW

**Comment by Grayman7000** [comment]
can't go wrong with a little go go gadget. What's the limit on size? Tiny? And how many items can you store in there?

**Comment by mc-oofert** [comment]
> can't go wrong with a little go go gadget. What's the limit on size? Tiny? And how many items can you store in there?

4 up to normal sized items

File: code/modules/clothing/head/jobs.dm [hunk 1]
```diff
@@ -206,6 +206,106 @@
...
+		wearer.visible_message(span_warning("[src] drops [result] into the hands of [wearer]!"))
+		items_by_phrase -= the_phrase
+	else
+		balloon_alert(wearer, "cant put in hands!")
+
+	return TRUE
+
+/obj/item/clothing/head/fedora/inspector_hat/attackby(obj/item/item, mob/user, params)
+	. = ..()
+	if(LAZYLEN(contents) >= max_weight)
```
Comment (Ghommie) [inline code/modules/clothing/head/jobs.dm:266]:
```suggestion
	if(LAZYLEN(contents) >= max_items)
```

File: code/modules/clothing/head/jobs.dm [hunk 2]
```diff
@@ -206,6 +206,106 @@
...
+		to_chat(user, span_warning("[src] is full!"))
+		return
+	if(item.w_class > WEIGHT_CLASS_NORMAL)
```
Comment (Ghommie) [inline code/modules/clothing/head/jobs.dm:269]:
```suggestion
	if(item.w_class > max_weight)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
You should add a definition of `Exited()` that tries to remove the exited movable from the item_by_phrase list instead of doing it manually each time the item is put in your hand.

File: code/modules/clothing/head/jobs.dm (hunk 2 above)
Comment (mc-oofert) [reply code/modules/clothing/head/jobs.dm:269]: oops

File: code/modules/clothing/head/jobs.dm [hunk 3]
```diff
@@ -206,6 +206,106 @@
...
+		to_chat(user, span_warning("[src] is full!"))
+		return
+	if(item.w_class > max_weight)
+		to_chat(user, span_warning("[item] is too big!"))
+		return
+	var/input = tgui_input_text(user, "What is the activation phrase?", "Activation phrase", "gadget", max_length = 26)
+	if(!input)
+		return
+	if(input in items_by_phrase)
+		to_chat(user, "[input] is already used!")
```
Comment (MrMelbert) [inline code/modules/clothing/head/jobs.dm:276]:
```suggestion
	if(input in items_by_phrase)
		to_chat(user, "[input] is already used!")
		return
```

File: code/modules/clothing/head/jobs.dm [hunk 4]
```diff
@@ -206,6 +206,106 @@
...
+	return TRUE
+
+/obj/item/clothing/head/fedora/inspector_hat/attackby(obj/item/item, mob/user, params)
+	. = ..()
+	if(LAZYLEN(contents) >= max_items)
+		to_chat(user, span_warning("[src] is full!"))
+		return
+	if(item.w_class > max_weight)
+		to_chat(user, span_warning("[item] is too big!"))
+		return
```
Comment (MrMelbert) [inline code/modules/clothing/head/jobs.dm:271]: All of these short to_chats should be balloon alerts

**Comment by mc-oofert** [comment]
all reviews as of now should be addressed

File: code/modules/clothing/head/jobs.dm (hunk 4 above)
Comment (mc-oofert) [reply code/modules/clothing/head/jobs.dm]: done

**Comment by Serxule** [comment]
cant wait for "go go gadget .357 revolver"

**Comment by Jordie0608** [comment]
It appears a fedora could fit another; go go gadget inspector's fedora.

**Comment by mc-oofert** [comment]
> It appears a fedora could fit another; go go gadget inspector's fedora.

theres only 1 per detectives cabinet and nowhere else

**Comment by optimumtact** [comment]
Your PR fails to mention how people get it.

**Comment by san7890** [comment]
> Your PR fails to mention how people get it.

changelog says its in the det's cabinet, unless i misunderstand

**Comment by Loafin34** [comment]
emagging it should let you store one bulky item in there. "go go gadget fire axe"

**Comment by mc-oofert** [comment]
> emagging it should let you store one bulky item in there. "go go gadget fire axe"

bulky items include a LOT of other undesirable stuff that could cause issues

**Comment by Loafin34** [comment]
> > emagging it should let you store one bulky item in there. "go go gadget fire axe"
> 
> bulky items include a LOT of other undesirable stuff that could cause issues

shit... damn... maybe theres a check for if there's anything stored in the bulky object first?

**Comment by Ghommie** [comment]
> 
> bulky items include a LOT of other undesirable stuff that could cause issues

Go go gadget transfer tank valve.

**Comment by Serxule** [comment]
there could be a bulky item varient thats nukie/admin only, but yeah bulky would be way too OP and probably annoying to code

**Comment by Ghommie** [comment]
The var is already there, silly.

**Comment by Serxule** [comment]
> The var is already there, silly.

ok well i dont look at the code for prs so

File: code/modules/clothing/head/jobs.dm [hunk 5]
```diff
@@ -206,6 +206,117 @@
...
+/obj/item/clothing/head/fedora/inspector_hat/Initialize(mapload)
+	. = ..()
+	become_hearing_sensitive(ROUNDSTART_TRAIT)
+	QDEL_NULL(atom_storage)
+
+/obj/item/clothing/head/fedora/inspector_hat/examine(mob/user)
+	. = ..()
+	. += span_notice("You can put items inside, and get them out by saying a phrase, or using it in-hand!")
+	. += span_notice("The prefix is <b>[prefix]</b>, and you can change it with alt-click!")
+	. += ""
```
Comment (san7890) [inline code/modules/clothing/head/jobs.dm:235]: what's the point of this? do you just want a new line? just do
```suggestion
	. += span_notice("The prefix is <b>[prefix]</b>, and you can change it with alt-click!\n")
```

this should work:tm: but let me know if i'm having a skill issue

Comment (mc-oofert) [reply code/modules/clothing/head/jobs.dm:235]: i pray that this works

**Review by Ghommie — APPROVED** [review]

File: code/modules/clothing/head/jobs.dm [hunk 6]
```diff
@@ -206,6 +206,116 @@
...
+		return
+	if(input in items_by_phrase)
+		balloon_alert(user, "already used!")
+		return
+
+	if(item.loc != user || !user.transferItemToLoc(item, src))
+		return
+
+	to_chat(user, span_notice("You install [item] into the [thtotext(contents.len)] slot in [src]."))
+	playsound(src.loc, 'sound/machines/click.ogg', 30, TRUE)
```
Comment (MrMelbert) [inline code/modules/clothing/head/jobs.dm:283]:
```suggestion
	playsound(src, 'sound/machines/click.ogg', 30, TRUE)
```
Same thing

**Comment by ZeWaka** [comment]
does not seem to be stolen code

**Pull request merged**